src/parsers/perf/perfparser.cpp

```diff
@@ -1342,6 +1369,10 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
                         emit bottomUpDataAvailable(d.bottomUpResult);
                         emit topDownDataAvailable(d.topDownResult);
                         emit summaryDataAvailable(d.summaryResult);
+
+                        d.disassemblyResult.copy(m_disassemblyResult);
```

Comment:
why is there both `d.disassemblyResult` and `m_disassemblyResult`?

Reply:
and this, afaik if you just keep the arch you can remove this one and all the strange functions on the disassembly result struct itself